flink-table-store-connector/src/main/java/org/apache/flink/table/store/connector/source/FlinkSourceBuilder.java

```diff
@@ -150,20 +153,20 @@ private FileStoreSource buildFileSource(boolean isContinuous, boolean continuous
 
             LogStartupMode startupMode = conf.get(LOG_SCAN);
             if (logSourceProvider == null) {
-                return buildFileSource(true, startupMode == LogStartupMode.LATEST);
+                return buildFileSource(true, startupMode, conf.get(LOG_SCAN_TIMESTAMP_MILLS));
             } else {
                 if (startupMode != LogStartupMode.FULL) {
                     return logSourceProvider.createSource(null);
                 }
                 return HybridSource.<RowData, StaticFileStoreSplitEnumerator>builder(
-                                buildFileSource(false, false))
+                                buildFileSource(false, startupMode, null))
                         .addSource(
                                 new LogHybridSourceFactory(logSourceProvider),
                                 Boundedness.CONTINUOUS_UNBOUNDED)
                         .build();
             }
         } else {
-            return buildFileSource(false, false);
+            return buildFileSource(false, LogStartupMode.FULL, null);
```

Comment:
Assert that startup mode is `FULL`.

Reply:
Here should ignore it. `log.scan` should not affect batch mode.